### ISSUE TITLE
It would be better to save the features if upstream is not trainable

### ISSUE THREAD

**Issue by raotnameh** [s3prl/s3prl#390]
Calculating the SAME features in each epoch is a redundant task and slows down the whole process significantly.

**Comment by leo19941227** [comment]
You are right. However, dumping all layers of feature for hubert or wav2vec2 requires huge amount of disk space. Also, I have tried that HDD would be a serious IO bottleneck, hence you must dump in SSD. All layers of hubert feature is about 70 times of the original dataset. Hence, when considering LibriSpeech, you need SSD space for 71 Librispeech for a single SSL representation, which can be kind of prohibitive.

However, if we only consider the few-shot finetuning setting, this can be quite a good suggestion and we should be option to switch between on-the-fly / pre-extracted mode. Thanks for the suggestion, will support in the future.

**Comment by raotnameh** [comment]
We can chunk the data and then load a part of it in RAM as required. 
Let's say 1000 data points at once and then repeat the process. 

Maybe if no saved features path is given, it will save them in a temporary folder and use them in the next epochs and delete the folder once the training is finished. 

Also, I can initiate a PR with this feature if you can guide me on what/where changes need to be done to reflect the task.

**Comment by raotnameh** [comment]
I have created a PR for the same. 
 https://github.com/s3prl/s3prl/pull/391#issue-1349503110
